### PR TITLE
refactor(share/sha-256): Consolidate SHA-256 Hash Function Usage 

### PR DESCRIPTION
**Pull request by kien6034**
## Closes 
Closes https://github.com/celestiaorg/celestia-node/issues/3275

## Notes 
This PR consolidates SHA-256 hash function into the `share/hashutils.go`, and replaces all the `sha256.New()` functions with `share.NewSHA256Hasher()`

### CODE REVIEW

File: share/hashutils.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import (
+	"crypto/sha256"
+	"hash"
+)
+
+// NewSHA256Hasher returns a new instance of a SHA-256 hasher.
+func NewSHA256Hasher() hash.Hash {
+	return sha256.New()
+}
```
Comment (walldiss) [inline share/hashutils.go:11]: It feels a bit overkill to have go file just for this function. Do you mind to move it to share.go?

Comment (kien6034) [reply share/hashutils.go]: not at all. I moved it to `share.go`!

**Review by walldiss — APPROVED** [review]
Thank you for contribution. LGTM

**Comment by kien6034** [comment]
@walldiss hi, i've resolve conflict with main branch, which dismissed your review. Can you please review this PR again! Thanks a alot

**Review by walldiss — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**